control-plane/connect-inject/peering_dialer_controller.go

```diff
@@ -55,112 +54,106 @@ func (r *PeeringDialerController) Reconcile(ctx context.Context, req ctrl.Reques
-	peering, _, err := r.ConsulClient.Peerings().Read(ctx, peeringDialer.Name, nil)
-	if err != nil {
-		r.Log.Error(err, "failed to get Peering from Consul", "name", req.Name)
-		return ctrl.Result{}, err
+	// Get the status secret and the spec secret.
+	secretRefSet := false
+	if dialer.SecretRef() != nil {
+		secretRefSet = true
+	}
+
+	// statusSecret will be nil if the secret specified by the status doesn't exist.
+	var statusSecret *corev1.Secret
+	if secretRefSet {
+		statusSecret, err = r.getSecret(ctx, dialer.SecretRef().Name, dialer.Namespace)
+		if err != nil {
+			r.updateStatusError(ctx, dialer, err)
+			return ctrl.Result{}, err
+		}
 	}
-	peeringExists := peering != nil
-	// TODO(peering): Verify that the existing peering in Consul is an dialer peer. If it is an acceptor peer, an error should be thrown.
 
 	// At this point, we know the spec secret exists. If the status secret doesn't
 	// exist, then we want to initiate peering and update the status with the secret for the token being used.
-	if existingStatusSecret == nil {
+	if statusSecret == nil {
 		// Whether the peering exists in Consul or not we want to initiate the peering so the status can reflect the
 		// correct secret specified in the spec.
-		r.Log.Info("status.secret doesn't exist or wasn't set, establishing peering with the existing spec.peer.secret", "secret-name", peeringDialer.Spec.Peer.Secret.Name, "secret-namespace", peeringDialer.Namespace)
-		peeringToken := existingSpecSecret.Data[peeringDialer.Spec.Peer.Secret.Key]
-		_, err := r.initiatePeering(ctx, peeringDialer.Name, string(peeringToken))
-		if err != nil {
-			_ = r.updateStatusError(ctx, peeringDialer, err)
+		r.Log.Info("status.secret doesn't exist or wasn't set, establishing peering with the existing spec.peer.secret", "secret-name", dialer.Secret().Name, "secret-namespace", dialer.Namespace)
+		peeringToken := specSecret.Data[dialer.Secret().Key]
+		if err := r.initiatePeering(ctx, dialer.Name, string(peeringToken)); err != nil {
+			r.updateStatusError(ctx, dialer, err)
 			return ctrl.Result{}, err
 		} else {
-			err := r.updateStatus(ctx, peeringDialer, existingSpecSecret.ResourceVersion)
+			err := r.updateStatus(ctx, dialer, specSecret.ResourceVersion)
 			return ctrl.Result{}, err
 		}
 	} else {
 		// At this point, the status secret does exist.
 		// If the peering in Consul does not exist, initiate peering.
-		if !peeringExists {
-			r.Log.Info("status.secret exists, but the peering doesn't exist in Consul; establishing peering with the existing spec.peer.secret", "secret-name", peeringDialer.Spec.Peer.Secret.Name, "secret-namespace", peeringDialer.Namespace)
-			peeringToken := existingSpecSecret.Data[peeringDialer.Spec.Peer.Secret.Key]
-			_, err := r.initiatePeering(ctx, peeringDialer.Name, string(peeringToken))
-			if err != nil {
-				_ = r.updateStatusError(ctx, peeringDialer, err)
+
+		// Read the peering from Consul.
+		// TODO(peering): do we need to pass in partition?
```

Comment:
Would this not already be set inside the `ConsulClient` ?

Reply:
Yup it should be, it'll use the partition of the local client agent which is fine because we do want to use the local partition we're in for the controllers. It was a leftover comment from when I wasn't 100% sure. So I think this comment can be removed!